### ISSUE TITLE
docker image not working on Raspi4 aka arm64

### ISSUE THREAD

**Issue by johanneskastl** [vadimkim/cert-manager-webhook-hetzner#18]
First off: Big thank you for this! It is working like a charm with three of my clusters.

The fourth kubernetes "cluster" is on a Raspberry Pi 4, where the image does not work. Apparently the image is built for amd64 only, while the Raspi is arm64.

I have no clue on how to build docker images with multiple architectures, so I cannot help. But it would be really nice if this would work on arm64, too.

Thanks in advance!

**Comment by johanneskastl** [comment]
Hmm, from what I found on go cross-compiling the only thing to change would be this:

```
RUN CGO_ENABLED=0 go build -o webhook -ldflags '-w -extldflags "-static"' .
```

TO

```
RUN CGO_ENABLED=0 GOARCH=arm64 go build -o webhook -ldflags '-w -extldflags "-static"' .
```

How to publish that is another matter...

**Comment by johanneskastl** [comment]
> Hmm, from what I found on go cross-compiling the only thing to change would be this:
> 
> ```
> RUN CGO_ENABLED=0 go build -o webhook -ldflags '-w -extldflags "-static"' .
> ```
> 
> TO
> 
> ```
> RUN CGO_ENABLED=0 GOARCH=arm64 go build -o webhook -ldflags '-w -extldflags "-static"' .
> ```

I copied the Dockerfile to `Dockerfile_arm64`, changed that line and built the image using `podman build -f Dockerfile_arm64`. Compilation was successful.

**Comment by johanneskastl** [comment]
> In my opinion the best approach would be to add a multi-platform image.

+1

> Used `docker buildx` to create it but cannot test it right now, would you mind: https://hub.docker.com/r/dgiebert/cert-manager-webhook-hetzner/tags

Looks promising, instead of the "exec format error" I get another one that at least shows the executable is running:
```
panic: GROUP_NAME must be specified
```
(I just ran this using `kubectl run webhook-test --image dgiebert/cert-manager-webhook-hetzner:latest`)

Can you outline the steps you used to build it? I tried to yesterday but got stuck in the whole buildah podman skopeo docker confusion and did only get a non working one...

**Comment by dgiebert** [comment]
> Can you outline the steps you used to build it? I tried to yesterday but got stuck in the whole buildah podman skopeo docker confusion and did only get a non working one...

The docs for this are here: https://docs.docker.com/buildx/working-with-buildx/#build-multi-platform-images

**Comment by vadimkim** [comment]
Building process works, but binary that goes into container must also be compiled for corresponding architecture. It is complicated. As a result image is broken. Multi-stage build can probably solve the problem, but it is complicated. Not a priority right now. Have to upgrade base image first

**Comment by vadimkim** [comment]
I managed to compile images for arm64 and arm7. Published at docker hub under version 1.2.0.  Can look it here:
https://hub.docker.com/layers/zmejg/cert-manager-webhook-hetzner/1.2.0/images/sha256-35b4511f3adbc92681e88ce94a7baf442d0e67329338a76cb36847b98d0e3104?context=explore

If you change value in deployment yaml to 1.2.0 it should pull correct version. Before I put "latest" tag and release new image at Helm you can test. I will release 1.2.0 in upcoming several days. It is build on 1.19.5 go template and alpine 3.17
